lib/dfe/analytics.rb

```diff
@@ -192,16 +192,26 @@ def self.models_for_entity(entity)
     def self.extract_model_attributes(model, attributes = nil)
       # if no list of attrs specified, consider all attrs belonging to this model
       attributes ||= model.attributes
-      table_name = model.class.table_name
+      table_name = model.class.table_name.to_sym
 
-      exportable_attrs = allowlist[table_name.to_sym].presence || []
-      pii_attrs = allowlist_pii[table_name.to_sym].presence || []
+      exportable_attrs = (allowlist[table_name].presence || []).map(&:to_sym)
+      hidden_pii_attrs = (hidden_pii[table_name].presence || []).map(&:to_sym)
+      pii_attrs = (allowlist_pii[table_name].presence || []).map(&:to_sym)
+
+      # Validation in fields.rb ensures attributes do not appear on both allowlist_pii and allowlist_hidden_pii
       exportable_pii_attrs = exportable_attrs & pii_attrs
+      exportable_hidden_pii_attrs = exportable_attrs & hidden_pii_attrs
+
+      # Exclude both pii and hidden attributes from allowed_attributes
+      allowed_attrs_to_include = exportable_attrs - (exportable_pii_attrs + exportable_hidden_pii_attrs)
 
-      allowed_attributes = attributes.slice(*exportable_attrs&.map(&:to_s))
-      obfuscated_attributes = attributes.slice(*exportable_pii_attrs&.map(&:to_s))
+      allowed_attributes = attributes.slice(*allowed_attrs_to_include&.map(&:to_s))
+      obfuscated_attributes = attributes.slice(*exportable_pii_attrs.map(&:to_s))
+                .transform_values { |value| pseudonymise(value) }
+      hidden_attributes = attributes.slice(*exportable_hidden_pii_attrs&.map(&:to_s))
 
-      allowed_attributes.deep_merge(obfuscated_attributes.transform_values { |value| pseudonymise(value) })
+      # Allowed attributes (which currently includes the allowlist_pii) must be kept separate from hidden_attributes
+      [allowed_attributes.deep_merge(obfuscated_attributes), hidden_attributes]
```

Comment:
Not sure if I have misunderstood, but I feel we can avoid a lot of changes if we return like:

``` Ruby 
{
  data: allowed_attributes.deep_merge(obfuscated_attributes),
  hidden_data: hidden_attributes
}
```